### DISCUSSION TITLE
Two mice for Lemmings

### DISCUSSION THREAD

**Discussion by boogenheimer** [BlitterStudio/amiberry#840] [Q&A] [answered]
I am trying to set up a kiosked version of Lemmings with AmiBerry on RetroPie. The game runs beautifully but I'm wondering if there is any way to set up a second mouse as a controller to allow for two player games? I have tried various things but cannot seem to make anything stick.

Secondly, for kiosking reasons I am trying to disable F12 (GUI toggle) and F10 (quit game) keys on the keyboard (needed to enter level codes) but I'm not having any luck. I've left the "default_open_gui_key=" entry blank in amiberry.conf but F12 still works. I cannot find an entry for F10, I assume this is a game specific option and cannot be disabled via the amiberry settings.

Any tips would be greatly appreciated.

**Comment by midwan** [comment, marked as answer]
Amiberry can support that, but SDL2 currently has support for only 1 mouse in the system.
Until that changes, there's not much I can do I'm afraid. I believe it's in their roadmap for the future...

F12 is the default key for Open GUI, but it can be changed to something else with the option you saw in the `amiberry.conf` file. If no option is specified there, the default applies. F10 is a game specific one, it's not intercepted by the emulator.
Keep in mind you can also use key combinations (e.g. Ctrl+F12) or game controller buttons, if you want.

**Reply by boogenheimer** [reply]
Ok understood, thanks for clarifying. I'm wondering if I can cheat the system by adding an old serial mouse maskerading as a joystick/controller?

It's a strange one really, most if not all posts relating to this topic are from people trying to make certain keystrokes/hotkeys work, whereas I'm trying to do the opposite and remove some of the functionality, so that the game cannot be exited by visitors. Is there a way to get the emulator to intercept key strokes from the keyboard? Or can the Open GUI command be mapped to a non-existent button on a non-existent controller?

**Reply by midwan** [reply]
Even a serial mouse would need to be handled by SDL2 as a 2nd mouse, and that's where the blocker is. Until they add support for it, we're kind of stuck in this.
You could assign the open GUI command to any key or gamepad button, by editing the config file manually, if you want.

**Reply by andyc1864** [reply]
Hello Midwan,

I have been in contact with a guy called Ryan who works on SDL. He has indicated that the option for introducing the feature to support two mice should be relatively easy to implement, but they have no plans to do so yet.

He did however say that if anyone were willing to come along with a patch he would be happy to support it through version control.

I don't have the coding skills to do that. Do you think it would be worth reaching out? Having the ability to support two mice for games such as Lemmings and Hired Guns would be awesome!

**Reply by midwan** [reply]
@andyc1864 
From what I understand, they did implement a piece at least towards that direction, but the solution is not complete yet. Check the related issue here, where they mention a commit added to support multi-mouse support: https://github.com/libsdl-org/SDL/issues/6223#issuecomment-1250239884

So it seems like they are (?) moving towards that direction already.

**Reply by andyc1864** [reply]
Any chance I can PM you and share the SDL email response to my query?

Thanks.

**Reply by midwan** [reply]
@andyc1864 sure, you can either mail me at midwan at gmail dot com, or find me on Discord (I'm available on multiple servers) or other channels (FB, Signal, etc)
